### PR TITLE
feat: [CDS-54157]: add instances checkbox in asg bg deploy step

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

> Added instances checkbox in ASG Blue Green Deploy Step

#### Screenshots

https://user-images.githubusercontent.com/106532291/224000845-4083d451-ae1e-4988-917f-53d1e4f36936.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/AsgBlueGreenDeployStep/AsgBlueGreenDeployStepEdit.tsx [hunk 1]
```diff
@@ -364,6 +368,17 @@ const AsgBlueGreenDeployStepEdit = (
                 stepViewType={stepViewType}
               />
 
+              <div className={stepCss.divider} />
+              <Text margin={{ bottom: 'medium' }}>{getString('instanceFieldOptions.instances')}</Text>
+              <div className={cx(stepCss.formGroup, stepCss.md)}>
+                <FormMultiTypeCheckboxField
```
Comment (PratyushGar) [inline src/modules/75-cd/components/PipelineSteps/AsgBlueGreenDeployStep/AsgBlueGreenDeployStepEdit.tsx:374]: With the current md styling text `Same as already running Instance`(s) is cropped

File: src/modules/75-cd/components/PipelineSteps/AsgBlueGreenDeployStep/AsgBlueGreenDeployStepInputSet.tsx [hunk 2]
```diff
@@ -327,6 +328,20 @@ const AsgBlueGreenDeployStepInputSet = (props: AsgBlueGreenDeployStepInputSetPro
...
       )}
+      {getMultiTypeFromValue(template?.spec?.useAlreadyRunningInstances) === MultiTypeInputType.RUNTIME && (
+        <div className={cx(stepCss.formGroup, stepCss.md)}>
+          <FormMultiTypeCheckboxField
+            multiTypeTextbox={{
+              expressions,
+              allowableTypes
+            }}
+            name={`${prefix}spec.useAlreadyRunningInstances`}
+            label={'Use Already Running Instances'}
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/AsgBlueGreenDeployStep/AsgBlueGreenDeployStepInputSet.tsx:339]: Add to strings

File: src/modules/75-cd/components/PipelineSteps/AsgBlueGreenDeployStep/AsgBlueGreenDeployStepEdit.tsx [hunk 3]
```diff
@@ -364,6 +368,17 @@ const AsgBlueGreenDeployStepEdit = (
                 stepViewType={stepViewType}
               />
 
+              <div className={stepCss.divider} />
+              <Text margin={{ bottom: 'medium' }}>{getString('instanceFieldOptions.instances')}</Text>
```
[the remark this answers is not in the document]
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineSteps/AsgBlueGreenDeployStep/AsgBlueGreenDeployStepEdit.tsx:372]: Yes this is desired

File: src/modules/75-cd/components/PipelineSteps/AsgBlueGreenDeployStep/AsgBlueGreenDeployStepInputSet.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineSteps/AsgBlueGreenDeployStep/AsgBlueGreenDeployStepInputSet.tsx:339]: Done

File: src/modules/75-cd/components/PipelineSteps/AsgBlueGreenDeployStep/AsgBlueGreenDeployStepEdit.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineSteps/AsgBlueGreenDeployStep/AsgBlueGreenDeployStepEdit.tsx:374]: Increased the width to lg

**Review by Eric761 — APPROVED** [review]

**Review by PratyushGar — DISMISSED** [review]

**Review by rameshx — DISMISSED** [review]

**Review by Eric761 — APPROVED** [review]

**Pull request merged**